### PR TITLE
RYA-440 Added commands to Rya Shell used to interact with Rya Streams.

### PR DESCRIPTION
**Pull request by kchilton2**
## Description
Integrated the Rya Shell with Rya Streams.

### Tests
Added tests.

### Links
https://issues.apache.org/jira/browse/RYA-440

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/673/

File: extras/indexing/src/main/java/org/apache/rya/api/client/accumulo/AccumuloSetRyaStreamsConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import org.apache.rya.api.client.InstanceExists;
+import org.apache.rya.api.client.SetRyaStreamsConfiguration;
+import org.apache.rya.api.client.SetRyaStreamsConfigurationBase;
+import org.apache.rya.api.instance.RyaDetailsRepository;
+
+import edu.umd.cs.findbugs.annotations.DefaultAnnotation;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * A Accumulo implementation of {@link SetRyaStreamsConfiguration}.
```
Comment (isper3at) [inline extras/indexing/src/main/java/org/apache/rya/api/client/accumulo/AccumuloSetRyaStreamsConfiguration.java]: nit: An*

File: extras/indexing/src/main/java/org/apache/rya/api/client/mongo/MongoSetRyaStreamsConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            final InstanceExists instanceExists,
+            final MongoClient client) {
+        super(instanceExists);
+        this.client = requireNonNull(client);
+    }
+
+    @Override
+    protected RyaDetailsRepository getRyaDetailsRepo(final String ryaInstance) {
+        requireNonNull(ryaInstance);
+        return new MongoRyaInstanceDetailsRepository(client, ryaInstance);
```
Comment (isper3at) [inline extras/indexing/src/main/java/org/apache/rya/api/client/mongo/MongoSetRyaStreamsConfiguration.java]: you could one line this?

File: extras/indexing/src/test/java/org/apache/rya/api/client/accumulo/AccumuloSetRyaStreamsConfigurationIT.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        final RyaClient ryaClient = AccumuloRyaClientFactory.build(connectionDetails, getConnector());
+
+        // Install an instance of Rya.
+        final Install installRya = ryaClient.getInstall();
+        final InstallConfiguration installConf = InstallConfiguration.builder()
+                .build();
+        installRya.install(ryaInstance, installConf);
+
+        // Fetch its details and show they do not have any RyaStreamsDetails.
+        com.google.common.base.Optional<RyaStreamsDetails> streamsDetails =
```
Comment (isper3at) [inline extras/indexing/src/test/java/org/apache/rya/api/client/accumulo/AccumuloSetRyaStreamsConfigurationIT.java]: we should really clean up the Optional stuff one day

File: extras/rya.streams/api/src/main/java/org/apache/rya/streams/api/interactor/defaults/DefaultGetQuery.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Get a {@link StreamsQuery} from Rya Streams.
+ */
+@DefaultAnnotation(NonNull.class)
+public class DefaultGetQuery implements GetQuery {
+    private final QueryRepository repository;
+
+    public DefaultGetQuery(final QueryRepository repository) {
```
Comment (isper3at) [inline extras/rya.streams/api/src/main/java/org/apache/rya/streams/api/interactor/defaults/DefaultGetQuery.java]: doc

File: extras/rya.streams/kafka/src/main/java/org/apache/rya/streams/kafka/KafkaRyaStreamsClientFactory.java [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+                }
+            }
+        };
+    }
+
+    /**
+     * Create a {@link Producer} that is able to write to a topic in Kafka.
+     *
+     * @param kafkaHostname - The Kafka broker hostname. (not null)
+     * @param kafkaPort - The Kafka broker port.
```
Comment (isper3at) [inline extras/rya.streams/kafka/src/main/java/org/apache/rya/streams/kafka/KafkaRyaStreamsClientFactory.java]: port restrictions?

File: extras/shell/src/test/java/org/apache/rya/shell/util/StreamsQueryFormatterTest.java [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+                        UUID.fromString("11111111-1111-1111-1111-111111111111"),
+                        "SELECT * WHERE { ?a ?b ?c . }",
+                        true),
+                new StreamsQuery(
+                        UUID.fromString("22222222-2222-2222-2222-222222222222"),
+                        "SELECT * WHERE { ?d ?e ?f . }",
+                        false));
+
+        final String formatted = StreamsQueryFormatter.format(queries);
+        System.out.println(formatted);
```
Comment (isper3at) [inline extras/shell/src/test/java/org/apache/rya/shell/util/StreamsQueryFormatterTest.java]: SOP?

File: extras/indexing/src/test/java/org/apache/rya/api/client/accumulo/AccumuloSetRyaStreamsConfigurationIT.java (hunk 3 above)
Comment (kchilton2) [reply extras/indexing/src/test/java/org/apache/rya/api/client/accumulo/AccumuloSetRyaStreamsConfigurationIT.java]: Google Optional objects are Java serializable. I'm pretty sure that's why they're being used.

File: extras/rya.streams/kafka/src/main/java/org/apache/rya/streams/kafka/KafkaRyaStreamsClientFactory.java (hunk 5 above)
Comment (kchilton2) [reply extras/rya.streams/kafka/src/main/java/org/apache/rya/streams/kafka/KafkaRyaStreamsClientFactory.java]: I'd rather just let it throw an exception.

File: extras/indexing/src/main/java/org/apache/rya/api/client/mongo/MongoSetRyaStreamsConfiguration.java (hunk 2 above)
Comment (kchilton2) [reply extras/indexing/src/main/java/org/apache/rya/api/client/mongo/MongoSetRyaStreamsConfiguration.java]: I think that makes the line look more congested when it isn't an assignment line.

File: extras/indexing/src/main/java/org/apache/rya/api/client/accumulo/AccumuloSetRyaStreamsConfiguration.java (hunk 1 above)
Comment (kchilton2) [reply extras/indexing/src/main/java/org/apache/rya/api/client/accumulo/AccumuloSetRyaStreamsConfiguration.java]: done

File: extras/rya.streams/api/src/main/java/org/apache/rya/streams/api/interactor/defaults/DefaultGetQuery.java (hunk 4 above)
Comment (kchilton2) [reply extras/rya.streams/api/src/main/java/org/apache/rya/streams/api/interactor/defaults/DefaultGetQuery.java]: done

File: extras/shell/src/test/java/org/apache/rya/shell/util/StreamsQueryFormatterTest.java (hunk 6 above)
Comment (kchilton2) [reply extras/shell/src/test/java/org/apache/rya/shell/util/StreamsQueryFormatterTest.java]: done

File: extras/indexing/src/main/java/org/apache/rya/api/client/mongo/MongoSetRyaStreamsConfiguration.java (hunk 2 above)
Comment (isper3at) [reply extras/indexing/src/main/java/org/apache/rya/api/client/mongo/MongoSetRyaStreamsConfiguration.java]: fair enough

File: extras/indexing/src/test/java/org/apache/rya/api/client/accumulo/AccumuloSetRyaStreamsConfigurationIT.java (hunk 3 above)
Comment (isper3at) [reply extras/indexing/src/test/java/org/apache/rya/api/client/accumulo/AccumuloSetRyaStreamsConfigurationIT.java]: yeah, still annoying we use 2

File: extras/rya.streams/kafka/src/main/java/org/apache/rya/streams/kafka/KafkaRyaStreamsClientFactory.java (hunk 5 above)
Comment (isper3at) [reply extras/rya.streams/kafka/src/main/java/org/apache/rya/streams/kafka/KafkaRyaStreamsClientFactory.java]: add a doc saying as much?

File: extras/indexing/src/test/java/org/apache/rya/api/client/accumulo/AccumuloSetRyaStreamsConfigurationIT.java (hunk 3 above)
Comment (kchilton2) [reply extras/indexing/src/test/java/org/apache/rya/api/client/accumulo/AccumuloSetRyaStreamsConfigurationIT.java]: You can open a ticket to change that if you have an improvement in mind.

File: extras/rya.streams/kafka/src/main/java/org/apache/rya/streams/kafka/KafkaRyaStreamsClientFactory.java (hunk 5 above)
Comment (kchilton2) [reply extras/rya.streams/kafka/src/main/java/org/apache/rya/streams/kafka/KafkaRyaStreamsClientFactory.java]: I don't think it's needed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/677/

File: extras/indexing/src/main/java/org/apache/rya/api/client/SetRyaStreamsConfigurationBase.java [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    /**
+     * Get a {@link RyaDetailsRepository} that is connected to a specific instance of Rya.
+     *
+     * @param ryaInstance - The Rya instance the repository must be connected to. (not null)
+     * @return A {@link RyaDetailsRepository} connected to the specified Rya instance.
+     */
+    protected abstract RyaDetailsRepository getRyaDetailsRepo(String ryaInstance);
+
+    @Override
+    public void setRyaStreamsConfiguration(final String ryaInstance, final RyaStreamsDetails streamsDetails) throws InstanceDoesNotExistException, RyaClientException{
```
Comment (meiercaleb) [inline extras/indexing/src/main/java/org/apache/rya/api/client/SetRyaStreamsConfigurationBase.java]: Null check on ryaInstance

File: extras/rya.streams/kafka/src/main/java/org/apache/rya/streams/kafka/KafkaRyaStreamsClientFactory.java [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+     * @param ryaInstance - The name of the Rya Instance the client is connected to. (not null)
+     * @param kafkaHostname - The hostname of the Kafka Broker.
+     * @param kafkaPort - The port of the Kafka Broker.
+     * @return The initialized commands.
+     */
+    public static RyaStreamsClient make(
+            final String ryaInstance,
+            final String kafkaHostname,
+            final int kafkaPort) {
+        requireNonNull(kafkaHostname);
```
Comment (meiercaleb) [inline extras/rya.streams/kafka/src/main/java/org/apache/rya/streams/kafka/KafkaRyaStreamsClientFactory.java]: Null check on ryaInstance

File: extras/indexing/src/main/java/org/apache/rya/api/client/SetRyaStreamsConfigurationBase.java (hunk 7 above)
Comment (kchilton2) [reply extras/indexing/src/main/java/org/apache/rya/api/client/SetRyaStreamsConfigurationBase.java]: done

File: extras/rya.streams/kafka/src/main/java/org/apache/rya/streams/kafka/KafkaRyaStreamsClientFactory.java (hunk 8 above)
Comment (kchilton2) [reply extras/rya.streams/kafka/src/main/java/org/apache/rya/streams/kafka/KafkaRyaStreamsClientFactory.java]: done

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/682/<h2>Failed Tests: <span class='status-failure'>2</span></h2><h3><a name='incubator-rya-master-with-optionals-pull-requests/org.apache.rya:rya.indexing' /><a href='https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/682/org.apache.rya$rya.indexing/testReport'>incubator-rya-master-with-optionals-pull-requests/org.apache.rya:rya.indexing</a>: <span class='status-failure'>2</span></h3><ul><li><a href='https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/682/org.apache.rya$rya.indexing/testReport/org.apache.rya.indexing.external.tupleSet/AccumuloIndexSetColumnVisibilityTest/org_apache_rya_indexing_external_tupleSet_AccumuloIndexSetColumnVisibilityTest/'><strong>org.apache.rya.indexing.external.tupleSet.AccumuloIndexSetColumnVisibilityTest.org.apache.rya.indexing.external.tupleSet.AccumuloIndexSetColumnVisibilityTest</strong></a></li><li><a href='https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/682/org.apache.rya$rya.indexing/testReport/org.apache.rya.indexing.external.tupleSet/AccumuloIndexSetColumnVisibilityTest/org_apache_rya_indexing_external_tupleSet_AccumuloIndexSetColumnVisibilityTest_2/'><strong>org.apache.rya.indexing.external.tupleSet.AccumuloIndexSetColumnVisibilityTest.org.apache.rya.indexing.external.tupleSet.AccumuloIndexSetColumnVisibilityTest</strong></a></li></ul>

**Pull request closed without merging**